kornia/geometry/transform/projwarp.py

```diff
@@ -130,3 +133,212 @@ def get_projective_transform(center: torch.Tensor, angles: torch.Tensor, scales:
+    if not src.shape == dst.shape:
+        raise ValueError("Inputs must have the same shape. Got {}"
+                         .format(dst.shape))
+    if not (src.shape[0] == dst.shape[0]):
+        raise ValueError("Inputs must have same batch size dimension. Expect {} but got {}"
+                         .format(src.shape, dst.shape))
+
+    def ax(p, q):
+        ones = torch.ones_like(p)[..., 0:1]
+        zeros = torch.zeros_like(p)[..., 0:1]
+        return torch.cat([
+            p[:, 0:1], p[:, 1:2], p[:, 2:3], ones,
+            zeros, zeros, zeros, zeros,
+            zeros, zeros, zeros, zeros,
+            -p[:, 0:1] * q[:, 0:1], -p[:, 1:2] * q[:, 0:1], -p[:, 2:3] * q[:, 0:1]
+        ], dim=1)
+
+    def ay(p, q):
+        ones = torch.ones_like(p)[..., 0:1]
+        zeros = torch.zeros_like(p)[..., 0:1]
+        return torch.cat([
+            zeros, zeros, zeros, zeros,
+            p[:, 0:1], p[:, 1:2], p[:, 2:3], ones,
+            zeros, zeros, zeros, zeros,
+            -p[:, 0:1] * q[:, 1:2], -p[:, 1:2] * q[:, 1:2], -p[:, 2:3] * q[:, 1:2]
+        ], dim=1)
+
+    def az(p, q):
+        ones = torch.ones_like(p)[..., 0:1]
+        zeros = torch.zeros_like(p)[..., 0:1]
+        return torch.cat([
+            zeros, zeros, zeros, zeros,
+            zeros, zeros, zeros, zeros,
+            p[:, 0:1], p[:, 1:2], p[:, 2:3], ones,
+            -p[:, 0:1] * q[:, 2:3], -p[:, 1:2] * q[:, 2:3], -p[:, 2:3] * q[:, 2:3]
+        ], dim=1)
+
+    # we build matrix A by using only 4 point correspondence. The linear
+    # system is solved with the least square method, so here
+    # we could even pass more correspondence
+    p = []
+
+    # 000, 100, 110, 101, 011
+    for i in [0, 1, 2, 5, 7]:
+        p.append(ax(src[:, i], dst[:, i]))
+        p.append(ay(src[:, i], dst[:, i]))
+        p.append(az(src[:, i], dst[:, i]))
+
+    # A is Bx15x15
+    A = torch.stack(p, dim=1)
+
+    # b is a Bx15x1
+    b = torch.stack([
+        dst[:, 0:1, 0], dst[:, 0:1, 1], dst[:, 0:1, 2],
+        dst[:, 1:2, 0], dst[:, 1:2, 1], dst[:, 1:2, 2],
+        dst[:, 2:3, 0], dst[:, 2:3, 1], dst[:, 2:3, 2],
+        # dst[:, 3:4, 0], dst[:, 3:4, 1], dst[:, 3:4, 2],
+        # dst[:, 4:5, 0], dst[:, 4:5, 1], dst[:, 4:5, 2],
+        dst[:, 5:6, 0], dst[:, 5:6, 1], dst[:, 5:6, 2],
+        # dst[:, 6:7, 0], dst[:, 6:7, 1], dst[:, 6:7, 2],
+        dst[:, 7:8, 0], dst[:, 7:8, 1], dst[:, 7:8, 2],
+    ], dim=1)
+
+    # solve the system Ax = b
+    X, LU = torch.solve(b, A)
+
+    # create variable to return
+    batch_size = src.shape[0]
+    M = torch.ones(batch_size, 16, device=src.device, dtype=src.dtype)
+    M[..., :15] = torch.squeeze(X, dim=-1)
+    return M.view(-1, 4, 4)  # Bx4x4
+
+
+def warp_perspective3d(src: torch.Tensor, M: torch.Tensor, dsize: Tuple[int, int, int],
+                       flags: str = 'bilinear', border_mode: str = 'zeros',
+                       align_corners: bool = False) -> torch.Tensor:
```

Comment:
have you verified that this is not the same a [`warp_projective`](https://kornia.readthedocs.io/en/latest/geometry.transform.html#kornia.geometry.transform.warp_projective) because it's the same idea. Maybe we should just rename to `warp_projective` -> `warp_perspective3d` or `warp_affine3d` (actually for the 3d case I'm not sure we can name either projective or even perspective transform. /cc @ducha-aiki

Reply:
They are different. I did not check the whole math under the hood. But intuitively, ```warp_projective``` accepts (B, 3, 4) matrix while ```warp_perspective3d``` accepts (B, 4, 4).

I would vote for renaming ```warp_projective ``` to ```warp_affine3d```. Logically, ```warp_projective``` aligns to ```warp_affine```, while ```warp_perspective3d``` aligns to ```warp_perspective```.

Reply:
@shijianjian I agree with warp_affine3d naming.

Reply:
@shijianjian can you verify if the transforms that you use for warp_perspective3d do they have last row 0001 ?

Reply:
It is not. For the example in doctest, the transformation matrix is:
```
tensor([[[ 1.6183, -0.0196,  0.6845,  0.3722],
         [-0.0692,  1.2327,  0.5895,  0.5762],
         [ 0.4182, -0.0778,  1.7113,  0.0664],
         [ 0.4492,  0.1442,  0.5956,  1.0000]]])
```